bin/bossSwitch.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python3.5
+
+# Copyright 2016 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script to turn the boss on and off.
+   The script aims at switch the boss on and off once it has been configured a first time.
+   The instances should first be stood up by building stacks through cloudformation.
+   ASG are shut down. Only execute this code if you are certain the boss will not
+   be running for another hour."""
+
+import boto3
+import sys
+import json
+import argparse
+import os
+import subprocess
+import time
+import pickle
+
+import alter_path
+from lib import aws, utils, vault
+
+def main():
+
+    choice = utils.get_user_confirm("Are you sure you want to proceed switching the boss on/off?")
+    if choice:
+        if args.action == "on":
+            print("Turning the BossDB on...")
+            startInstances()
+
+        elif args.action == "off":
+            print("Turning the BossDB off...")
+            stopInstances()
+
+#Executed actions
+def startInstances():
+    """
+        Method used to start necessary instances
+    """
+    #Use auto scaling groups last saved configuration
+    ASGdescription = load_obj('ASGdescriptions')
```

Comment:
Keep in mind that someone may manage multiple boss stacks that they want to turn off. Probably not common but there should be either a check to not override existing data or to create a unique save file per stack. Possibly just passing the config file as a command line parameter.